src/index/base.cpp

```diff
@@ -351,10 +360,12 @@ bool BaseIndex::IgnoreChainStateFlushed(const CBlockLocator& locator)
     // event, log a warning and let the queue clear.
     const CBlockIndex* best_block_index = m_best_block_index.load();
     if (best_block_index->GetAncestor(locator_tip_index->nHeight) != locator_tip_index) {
-        LogPrintf("%s: WARNING: Locator contains block (hash=%s) not on known best " /* Continued */
-                  "chain (tip=%s); not writing index locator\n",
-                  __func__, locator_tip_hash.ToString(),
-                  best_block_index->GetBlockHash().ToString());
+        if (m_synced) {
```

Comment:
f56a851e11321ede9c5bcb43cd067c17f9efab39: Adding the condition `if (m_synced)` seems unnecessary at this point, `IgnoreChainStateFlushed` returns right at the beginning if not synced.